packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts

```diff
@@ -437,6 +437,8 @@ export class OdspDocumentStorageService extends OdspDocumentStorageServiceBase {
 			const result = await this.cache.snapshotPrefetchResultCache
 				?.get(prefetchCacheKey)
 				?.then(async (response) => {
+					// Remove it from cache once used.
+					this.cache.snapshotPrefetchResultCache.remove(prefetchCacheKey);
```

Comment:
prefetchLatestSnapshot() that puts this cache entry will delete it as well. It feels like this should not be required, no?
I.e. what is the scenario where removing it here is required / changes the flow?

Reply:
@vladsud I described in the PR description and by comment in the prefetchSnapshot file. 
Scenario is that if we clear the cache immediately after caching it in persistent cache, then during load, when we fetch the snapshot parallely, then we won't find the snapshot in non persistant cache and make the network call again. So we will be making an extra Trees latest call. So that is why we are clearing it after 5s or once it used on load.